### PR TITLE
docs: update custom domain docs for verification process

### PR DESCRIPTION
**Pull request by Crispy1975**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?
Updating docs for custom domain verification to ensure the process and the docs match.

### CODE REVIEW

**Review by darora — APPROVED** [review]

File: apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainTableHeaders.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+export type DNSTableHeaderProps = {
+  display: string | undefined
```
Comment (alaister) [inline apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainTableHeaders.tsx:2]: Does this need to have `undefined` as an option? The `if` statement doesn't account for the `undefined` condition and where `<DNSTableHeaders />` is used has a string fallback anyway

Comment (alaister) [inline apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainTableHeaders.tsx:2]:
```suggestion
  display: string
```

Comment (Crispy1975) [reply apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainTableHeaders.tsx:2]: I'll see what I can do...

Comment (Crispy1975) [reply apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainTableHeaders.tsx]: Fixed. :-)

Comment (Crispy1975) [reply apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainTableHeaders.tsx]: Done.

File: apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainVerify.tsx [hunk 2]
```diff
@@ -31,18 +33,26 @@ const CustomDomainVerify = ({ projectRef, customDomain, settings }: CustomDomain
...
   )
+  const isValidating = (customDomain.ssl.txt_name ?? '') === ''
 
   const onReverifyCustomDomain = () => {
     if (!projectRef) return console.error('Project ref is required')
     reverifyCustomDomain({ projectRef })
   }
 
+  useInterval(
+    onReverifyCustomDomain,
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainVerify.tsx:51]: we can just set a refetchInterval for this - i don't think we need a separate hook?

**Review by joshenlim — APPROVED** [review]
LGTM! Tested locally and on staging adding custom domain to a project

**Review by alaister — APPROVED** [review]

**Pull request merged**